### DISCUSSION TITLE
GitHub Provider Support for GitHub App Authentication

### DISCUSSION THREAD

**Discussion by evandam** [fluxcd/flux2#4356] [Proposals]
Hey folks, thanks off the bat for an awesome product!

I was wondering if it would be reasonable to support using a GitHub App instead of a PAT? As explained in  https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/authenticating-as-a-github-app

While a PAT definitely works, it's a little funny seeing my GitHub username on every commit's checks, and it's a bit of a pain having to keep a central "bot user" in a GitHub organization. I'm sure there are some extra complications that come along with it, but curious what you all think 🙌 

Edit: maybe https://github.com/google/go-github#as-a-github-app helps, assuming this is the module being used.

**Comment by justinharringa** [comment]
I think this would be much better than creating a bot user that effectively takes a seat and requires MFA, etc...

**Comment by hiddeco** [comment]
This is being worked on as part of a bigger effort to also cache OIDC tokens, etc.

See https://github.com/fluxcd/pkg/pull/664 for the initial work and more information, which once merged will be implemented where applicable across the Flux sub-projects.
